Adds the Wikiann NER corpus for 8 Indonesian languages

kaustubhdhole opened the pull request:
Closes #138 "

The code has been enabled for selecting any language. The languages map LANGUAGES_MAP & LANG_CODES needs to be filled with other languages which are Indonesian.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `nusantara/nusa_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_NUSANTARA_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `NusantaraConfig` for the source schema and one for a nusantara schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [ ] Confirm that your dataloader script passes the test suite run with `python -m tests.test_nusantara --path=nusantara/nusa_datasets/my_dataset/my_dataset.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

holylovenia reviewed (changes requested):
Hi, @kaustubhdhole. Thank you for implementing WikiANN dataloader!

Could you help us by modifying a few things?
1. `_LANGUAGES` should use 3-letter ISO code, so please change it to `_LANGUAGES = ["ind", "eng"]`.
2. Could you please make the dataset name to `wikiann` instead of `wiki_ann`?
3. Maybe remove `self.` for `LANG_CODES` in line 144 so it won't throw an error?
4. Could we use `wikiann_ind_source` as the default schema config instead of `wiki_ann_source`?
5. The `description` in line 90 is not consistent with the dataloader's default behavior (if `lang` is empty). Maybe it's better to let the dataloader throw an error if the config name has the `lang` part empty. For the details of the dataloader's behavior, maybe you could check out PR #142 since it follows a similar workflow.

Please let me know if you need anything. :smile:

kaustubhdhole commented:
Thanks @holylovenia for the suggestions! Will update this PR shortly!

SamuelCahyawijaya reviewed (changes requested):
Hi @kaustubhdhole, thanks for contributing! I just checked the data manually, and somehow I get 0 instances for the validation set as shown below:
```
DatasetDict({
    validation: Dataset({
        features: ['index', 'tokens', 'ner_tag'],
        num_rows: 0
    })
    test: Dataset({
        features: ['index', 'tokens', 'ner_tag'],
        num_rows: 10000
    })
    train: Dataset({
        features: ['index', 'tokens', 'ner_tag'],
        num_rows: 20000
    })
})
```
but when I download the original data there is the _validation_ set (named as _dev_). In addition, I realize that there is an `extra` file on each language in the source data, could you add another split for that, I think you can just fill the name attribute for the split generator with _extra_.

In addition, I am sorry for the lack of description on the issue, but other than Indonesian and English, I checked https://elisa-ie.github.io/wikiann/ and they actually have the following Indonesian local languages (+ English):

| Language | WikiAnn | ISO 639-3 |
|---|---|---|
| Indonesian | id | ind |
| Javanese | jv | jav |
| Minangkabau | min | min |
| Sundanese | su | sun |
| Acehnese | ace | ace |
| Malay | ms | mly |
| Banyumasan | map-bms | map-bms |

Could you please help to implement the other languages as well? Thank you!

kaustubhdhole commented:
Thanks @SamuelCahyawijaya! I've fixed the issues you pointed out and added other languages as well. For  "Banyumasan", I've used "bms" as the code. The test case does not specify language so it's throwing an error. Should I default to one of the languages or leave it as it is?

SamuelCahyawijaya reviewed (changes requested):
Hi @kaustubhdhole, to run the unit test, you can use `--subset_id` as an additional parameter, for example:
```
python -m tests.test_nusantara nusantara/nusa_datasets/wikiann/wikiann.py --subset_id wikiann_ind
````
to run the unit test for `wikiann_ind_source` and `wikiann_ind_nusantara_seq_label`

I tested with that one and it passes the unit test. I think the PR looks good, except for the `bms` subset, since `bms` is used for Bilma Kanuri language. I think we can just use `wikiann_map_bms_nusantara_seq_label` and `wikiann_map_bms_source` for Banyumasan.

Thanks for the quick update!

kaustubhdhole commented:
Okay, I've kept the original lang tags then :)

SamuelCahyawijaya reviewed (approved):
Thanks for the update! Approving this PR 👍🏻

holylovenia reviewed (approved):
Thank you for implementing this dataloader and applying the changes, @kaustubhdhole! It works great. Merging now.
